### DISCUSSION TITLE
Detector rotations

### DISCUSSION THREAD

**Discussion by ttortorici** [silx-kit/pyFAI#2247] [General]
Hello,

So the description of the geometry is that rotations are done in numerical order: rot1, rot2, rot3. Rot1 is about the vertical axis making the detector move right. The axis of rotation for rot2 must be orthogonal to this axis, but my question is: is the axis of rotation for rot2 orthogonal to the direction of the PONI or to the direction of the beam after a rot1 rotation. My understanding is that rot3 is a rotation around the axis of the beam, so I'm leaning toward the latter choice, but I'm very unsure.

Thanks!

**Comment by kif** [comment]
Hi Teddy,

I did that a long time ago ... and forgot all about it except this scheme:
http://www.silx.org/doc/pyFAI/dev/_images/PONI.png
This is also discussed in the documentation at:
http://www.silx.org/doc/pyFAI/dev/geometry_conversion.html

The rotations are composed one on top of another but your question is whether the axes are modified by the previous rotation.
Rot1,2,3 are Euler angles and they could be defined as "sxyz" in the `transformation` library (from C. Gohlke), i.e. the rotation is always performed in the lab-space and not in the previously rotated space. 

This small snippet of code allows you to check easily:
```
import pyFAI
from pyFAI.gui import jupyter
from matplotlib.pyplot import subplots
from math import pi

ai = pyFAI.load({"dist":1,"detector": "titan", "rot1": pi/2, "rot2":pi/2, "rot3":0})
z,y,x = ai.calc_pos_zyx()

fig,ax = subplots(1,3, figsize=(12,4))
jupyter.display(x,ax=ax[0], label="x")
jupyter.display(y,ax=ax[1], label="y")
jupyter.display(z,ax=ax[2], label="z")
print(ai)
```
Provides `x` constant. This is consistent with static axis. Relative axis would provide `z` constant which is not what is observed.
